packages/smock/src/smockit/binding.ts

```diff
@@ -1,9 +1,22 @@
 /* Imports: External */
 import { HardhatNetworkProvider } from 'hardhat/internal/hardhat-network/provider/provider'
-import { decodeRevertReason } from 'hardhat/internal/hardhat-network/stack-traces/revert-reasons'
 import { VmError } from '@nomiclabs/ethereumjs-vm/dist/exceptions'
 import BN from 'bn.js'
 
+// Handle hardhat ^2.4.0
+let decodeRevertReason: (value: Buffer) => string
+try {
+  decodeRevertReason = require('hardhat/internal/hardhat-network/stack-traces/revert-reasons')
+    .decodeRevertReason
+} catch (err) {
+  const {
```

Comment:
Makes a lot of sense. I was wondering how you'd support both versions.

Reply:
Yeah meh this is good enough for now.